### PR TITLE
HMD UI Fixes

### PR DESCRIPTION
**Pull request by jherico**
* Don't automatically hide the Tool window when toggling the pinned state on the desktop
* Remove the head pose roll from the cursor transformation

### CODE REVIEW

**Comment by jherico** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;yTJ9neL0&#x2F;40-the-tool-window-sometimes-misbehaves-it-seems-to-shut-itself-off-when-the-desktop-toggles"> The tool window sometimes misbehaves. It seems to shut itself off when the desktop toggles.</a>

**Comment by jherico** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;wrTyaBBx&#x2F;74-the-cursor-is-not-supposed-to-roll-with-the-head"> The cursor is not supposed to roll with the head</a>

**Comment by hifi-gustavo** [comment]
Build console output for commit: 21d2c977a36f656b0a1c3c56afa435c8c1c5098f available at: https://gist.github.com/f779035ecbfa2db8c04bbf090494750b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8157.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8157.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 21d2c977a36f656b0a1c3c56afa435c8c1c5098f available at: https://gist.github.com/f779035ecbfa2db8c04bbf090494750b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8157.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8157.dmg

**Comment by jherico** [comment]
## Testing

* Tiling your head left and right while in the HMD should no longer cause the cursor to rotate around it's point.
* Pressing the edit button to enable edit mode, and the toggling the UI on and off with the button or by movement should no longer cause the edit tools window to stay hidden when the UI comes back.

File: libraries/shared/src/GLMHelpers.cpp [hunk 1]
```diff
@@ -466,6 +466,12 @@ glm::mat4 createMatFromScaleQuatAndPos(const glm::vec3& scale, const glm::quat&
                      glm::vec4(zAxis, 0.0f), glm::vec4(trans, 1.0f));
 }
 
+// cancel out roll 
+glm::quat cancelOutRoll(const glm::quat& q) {
+    glm::vec3 forward = q * glm::vec3(0.0f, 0.0f, -1.0f);
```
Comment (howard-stearns) [inline libraries/shared/src/GLMHelpers.cpp]: I realize that the vec3 "call" here didn't change from the previous code, but given the variable name change and the constants on the new next line, should this glm::vec3(0.0f, 0.0f, -1.0f) be changed to Vectors::FRONT?

File: interface/resources/qml/ToolWindow.qml [hunk 2]
```diff
@@ -137,12 +137,15 @@ ScrollingWindow {
     }
 
     function updateVisiblity() {
-        for (var i = 0; i < tabView.count; ++i) {
-            if (tabView.getTab(i).enabled) {
-                return;
+        if (visible) {
+            for (var i = 0; i < tabView.count; ++i) {
+                if (tabView.getTab(i).enabled) {
+                    console.log("QQQ tab is visible, returning early");
```
Comment (howard-stearns) [inline interface/resources/qml/ToolWindow.qml]: Is it intentional to leave this logging in? Do we we need it? Do we need the "QQQ" label?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5a9be817df1fc30898ca988efb5607150d8c93ce available at: https://gist.github.com/f779035ecbfa2db8c04bbf090494750b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8157.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8157.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** FAILURE
**Additional Test Results + Observations**:  On initial load of Interface, when switching to HMD mode, it cased a crash.  Log file included for further analysis to determine if it is unrelated to PR or something else.  After the single crash, I re-loaded Interface and it was in HMD mode.  Utilizing an Oculus, when I tilted my head left and right while in the HMD the cursor did in-fact rotate around it's point as I tilted left the cursor in a circular motion turned counter-clockwise on its axis.  Pressing the edit button to enable edit mode, and the toggling the UI on and off with the button or by movement did in fact no longer cause the edit tools window to stay hidden when the UI comes back.

**Comment by jherico** [comment]
@kevinmthomasse does the crash on switching to HMD mode a) happen reliably and only in this build, not in master and b) can you attach screenshots of the cursor rotated a bit clockwise and counterclockwise?

**Comment by kevinmthomasse** [comment]
## QA Re-Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** UNKNOWN
**Additional Test Results + Observations**:  The crash on switching to HMD mode absolutely is not reliable as I have not been able to reproduce it as I have tried on three additional separate occurences and was unabel to crash.  I have attached images of the ```tilted left``` and ```tilted right```.  Keep in mind the ```tilted right``` does not turn as much clockwise as the ```tilted left``` turns counter clockwise.

tilted left
tilted right

**Comment by jherico** [comment]
@kevinmthomasse OK, I'm seeing similar results in terms of rotation.  Unfortunately, that's not quite the problem I was trying to fix.  The cursor still rotates around it's point based on head orientation, but only with increasing intensity as the cursor approaches the poles.  I'll have to re-address this remaining issue in a future PR.   Previously, in production build, it would rotate even when you were looking straight ahead, and it should no longer do that.  In those terms, I believe this PR does what it was intending to.

**Comment by howard-stearns** [comment]
I concur. Pass.

**Pull request merged**